monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -1234,6 +1234,12 @@ sealed abstract class Task[+A] extends Serializable {
   final def foreach(f: A => Unit)(implicit s: Scheduler): CancelableFuture[Unit] =
     foreachL(f).runAsync(s)
 
+  /** Returns a new `Task` that executes the source repetitively.
+    *
+    */
+  final def forever: Task[A] =
+    flatMap(_ => this.forever)
```

Comment:
I think `loopForever` will read better when I see it in code. It should also probably return `Task[Nothing]` to indicate non-termination.

Reply:
Regarding `Task[Nothing]`, would have to use the async creator to get that type signature, since we can't produce a value of type `Nothing`, yeah? Could also use `Task[Unit]`, but I suppose that indicates termination, just with a value of type `Unit`.

Reply:
@RaasAhsan you can't produce a value, but you can have that in type if you never return anything.

E.g. this typechecks:
```
def loop: Nothing = loop
```

But if I *call* `loop`, I will never get a value back.

Reply:
Got it, that makes sense, just need to change the type signature then.